### PR TITLE
Devirtualize HashRouter

### PR DESCRIPTION
**Pull request by ximinez**
* Include some simple renames

Other than renaming a couple of things, and calling `make_unique` instead of `IHashRouter::New`, none of the code here is new, just moved.

Reviewers: @vinniefalco @miguelportilla

### CODE REVIEW

File: src/ripple/app/misc/HashRouter.h [hunk 1]
```diff
@@ -59,39 +121,56 @@ class IHashRouter
...
 
-    // VFALCO TODO rename the parameter to entryHoldTimeInSeconds
-    static IHashRouter* New (int holdTime);
+    explicit HashRouter (int entryHoldTimeInSeconds)
+        : mHoldTime (entryHoldTimeInSeconds)
+    {
+    }
 
-    virtual ~IHashRouter () { }
+    virtual ~HashRouter () { }
```
Comment (vinniefalco) [inline src/ripple/app/misc/HashRouter.h]: Could be defaulted

Comment (ximinez) [inline src/ripple/app/misc/HashRouter.h]: Done.

**Comment by vinniefalco** [comment]
Could be squashed

**Comment by ximinez** [comment]
Squashed.

**Comment by vinniefalco** [comment]
:+1:

Could be marked Passed

**Comment by ximinez** [comment]
Been rebased.

**Comment by vinniefalco** [comment]
Could be merged

**Comment by miguelportilla** [comment]
:+1:  Trivial change, builds and passes tests on all platforms. 

(Nothing to do with this PR but the annoying warning: ‘template<class> class std::auto_ptr’ is deprecated appear with GCC 5.x)

**Pull request closed without merging**